### PR TITLE
Remove long deprecated occurrences request

### PR DESCRIPTION
**Pull request by sheetalkamat**
Deprecated since https://github.com/microsoft/TypeScript/pull/22067

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Seems unused from what I can tell:

https://github.com/search?q=tsserver+%5C%22occurrences%5C%22+OR+%5C%27occurrences%5C%27+NOT+path%3A%2F%5Esrc%5C%2FtestRunner%5C%2Funittests%5C%2Ftsserver%5C%2F%2F+NOT+language%3AMarkdown+path%3A+NOT+path%3A%2F%5Esrc%5C%2Fserver%5C%2F%2F+NOT+path%3A%2F%5C.d%5C.ts%24%2F++NOT+path%3AtsserverProjectSystem.ts+NOT+path%3A.pnpm-store&type=code

File: src/harness/fourslashImpl.ts [hunk 1]
```diff
@@ -3497,7 +3497,16 @@ export class TestState {
     }
 
     private getOccurrencesAtCurrentPosition() {
-        return this.languageService.getOccurrencesAtPosition(this.activeFile.fileName, this.currentCaretPosition);
+        return ts.flatMap(
+            this.languageService.getDocumentHighlights(this.activeFile.fileName, this.currentCaretPosition, [this.activeFile.fileName]),
```
Comment (DanielRosenwasser) [inline src/harness/fourslashImpl.ts:3501]: There's some extra whitespace here

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
Code itself looks good to me, but @mjbvz and @zkat should probably sign off for VS Code and VS respectively.

**Comment by sandersn** [comment]
@sheetalkamat I'm going through all the ready-to-merge PRs, and this one seems worthwhile to merge. Given @jakebailey 's findings we might not need confirmation from @mjbvz or @zkat. What do you think?

**Comment by jakebailey** [comment]
This broke Monaco, because they use occurrences: https://github.com/microsoft/TypeScript-Make-Monaco-Builds/actions/runs/4308395883/jobs/7514631424

Can we revert this until we get a replacement into Monaco? We can't produce any playground builds at the moment.

**Comment by jakebailey** [comment]
I guess I can just send a PR with copy/paste of https://github.com/microsoft/TypeScript/pull/52347/files#diff-68e0a0bdd65b7a9bb405747188b445f5836ab24070c3c06501a961d5d69c910bR3500?

**Comment by jakebailey** [comment]
I sent https://github.com/microsoft/monaco-editor/pull/3619 so hopefully that's looked at soonish, but otherwise we might want to revert?

**Comment by sheetalkamat** [comment]
#53056 if we need to revert

**Pull request merged**